modulemd/v2/include/modulemd-2.0/private/modulemd-yaml.h

```diff
@@ -421,4 +421,129 @@ modulemd_yaml_emit_document_headers (yaml_emitter_t *emitter,
                                      guint64 mdversion,
                                      GError **error);
 
+
+/* A set of macros for simple emitting of common elements */
+#define NON_EMPTY_TABLE(table) g_hash_table_size (table) != 0
+
+#define EMIT_KEY(emitter, error, value)                                       \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_scalar (                                               \
+            emitter, value, YAML_PLAIN_SCALAR_STYLE, error))                  \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_SCALAR_IF_SET(emitter, error, key, value)                        \
+  do                                                                          \
+    {                                                                         \
+      if (value != NULL)                                                      \
+        {                                                                     \
+          EMIT_KEY (emitter, error, key);                                     \
+          if (!mmd_emitter_scalar (                                           \
+                emitter, value, YAML_PLAIN_SCALAR_STYLE, error))              \
+            return FALSE;                                                     \
+        }                                                                     \
+    }                                                                         \
+  while (0)
+
+#define EMIT_MAPPING_START(emitter, error)                                    \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_start_mapping (                                        \
+            emitter, YAML_BLOCK_MAPPING_STYLE, error))                        \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_MAPPING_END(emitter, error)                                      \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_end_mapping (emitter, error))                          \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_SEQUENCE_START(emitter, error)                                   \
```

Comment:
I wonder if we should handle the style here, since there are certain places in the emitters (the profile streams, for example) that we expressly use the FLOW style for. I suppose we can just treat those as exceptions, though.

Reply:
Personally, I feel making the default simplest.
To hopefully find a middle way, I added a `EMIT_SEQUENCE_START_WITH_STYLE` and `EMIT_MAPPING_START_WITH_STYLE` next to the default `EMIT_SEQUENCE_START` and `EMIT_MAPPING_START`, which use the former with block style.